src/pages/gallery/index.tsx

```diff
@@ -362,6 +370,72 @@ export default function Gallery() {
             });
         }
     };
+    const archiveFilesHelper = async () => {
+        loadingBar.current?.continuousStart();
+        try {
+            const archivedFiles = await changeFilesVisibility(
+                files,
+                selected,
+                VISIBILITY_STATE.ARCHIVED
+            );
+            await updateMagicMetadata(archivedFiles);
+        } catch (e) {
+            console.log(e);
+            switch (e.status?.toString()) {
+                case ServerErrorCodes.FORBIDDEN:
+                    setDialogMessage({
+                        title: constants.ERROR,
+                        staticBackdrop: true,
+                        close: { variant: 'danger' },
+                        content: constants.NOT_FILE_OWNER,
+                    });
+                    return;
+            }
+            setDialogMessage({
+                title: constants.ERROR,
+                staticBackdrop: true,
+                close: { variant: 'danger' },
+                content: constants.UNKNOWN_ERROR,
+            });
+        } finally {
+            clearSelection();
+            syncWithRemote();
+            loadingBar.current.complete();
+        }
+    };
+
+    const unArchiveFilesHelper = async () => {
```

Comment:
can merge both `archiveFilesHelper` and `unArchiveFilesHelper` to reduce duplication?